### PR TITLE
remove click area around "exit" button

### PR DESCRIPTION
**Pull request by undef21**
issue #2394

### CODE REVIEW

File: src/fheroes2/castle/castle_dialog.cpp [hunk 1]
```diff
@@ -270,6 +270,7 @@ int Castle::OpenDialog( bool readonly )
 
     // resource
     const Rect & rectResource = RedrawResourcePanel( cur_pt );
+    const Rect & resActiveArea = Rect( rectResource, rectResource.w, 165 );
```
Comment (ihhub) [inline src/fheroes2/castle/castle_dialog.cpp:273]: We should always hardcoded values. Why don't we create this area based on `buttonExit` position?

Comment (ihhub) [inline src/fheroes2/castle/castle_dialog.cpp:273]: Please also use `fheroes2::Rect` and a normal constructor in this situation.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @undef21 , could we please change the logic of calculating area for Income based on my suggestions?

File: src/fheroes2/castle/castle_dialog.cpp (hunk 1 above)
Comment (undef21) [reply src/fheroes2/castle/castle_dialog.cpp]: Done.

Comment (undef21) [reply src/fheroes2/castle/castle_dialog.cpp]: Done.

File: src/fheroes2/castle/castle_dialog.cpp [hunk 2]
```diff
@@ -325,11 +326,11 @@ int Castle::OpenDialog( bool readonly )
 
         le.MousePressLeft( buttonExit.area() ) ? buttonExit.drawOnPress() : buttonExit.drawOnRelease();
 
-        if ( le.MouseClickLeft( rectResource ) ) {
+        if ( le.MouseClickLeft( Rect( resActiveArea ) ) ) {
```
Comment (ihhub) [inline src/fheroes2/castle/castle_dialog.cpp:329]: Please remove an explicit class conversion. We're going to remove Rect class in the future.

Comment (undef21) [reply src/fheroes2/castle/castle_dialog.cpp]: Done.

**Comment by ihhub** [comment]
Hi @undef21 thanks a lot for fixing this issue!

**Pull request merged**